module/Company/src/Company/Controller/AdminController.php

```diff
@@ -0,0 +1,311 @@
+    {
+        $companyService = $this->getCompanyService();
+        $packetForm = $companyService->getPacketForm();
+        $request = $this->getRequest();
+        $companyName = $this->params('slugCompanyName');
+        if ($request->isPost()) {
+            $companyService = $this->getCompanyService();
+            $packetForm->setData($request->getPost());
+
+            if ($packetForm->isValid()) {
+                $packet = $companyService->insertPacketForCompanySlugName($companyName);
+                $packet->exchangeArray($request->getPost());
+                $companyService->savePacket();
+
+                return $this->redirect()->toRoute('admin_company/editCompany', array('slugCompanyName' => $companyName), array(), false);
+            }
+        }
+        $packetForm->setAttribute('action',
+                                   $this->url()->fromRoute('admin_company/editCompany/addPacket',
+                                                           array('slugCompanyName' => $companyName))
+                                  );
+        $vm = new ViewModel(array(
+          //  'company' => $company,
+            'companyEditForm' => $packetForm,
+        ));
+
+        return $vm;
+    }
+    public function addCompanyAction()
+    {
+        $companyService = $this->getCompanyService();
+        $companyForm = $companyService->getCompanyForm();
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            if (!isset($companyName)) {
+                $companyName = $request->getPost()['slugName'];
+            }
+            $companyService = $this->getCompanyService();
+            $companyForm = $companyService->getCompanyForm();
+            $companyForm->setData($request->getPost());
+
+            if ($companyForm->isValid()) {
+                $company = $companyService->insertCompany($request->getPost()['languages']);
+                $company->exchangeArray($request->getPost());
+                $companyService->saveCompany();
+
+                return $this->redirect()->toRoute('admin_company/default', array('action' => 'edit', 'slugCompanyName' => $companyName), array(), false);
+            }
+        }
+        $companyForm->setAttribute('action',
+                                   $this->url()->fromRoute('admin_company/default',
+                                                           array('action' => 'addCompany'))
+                                  );
+        $vm = new ViewModel(array(
+          //  'company' => $company,
+            'companyEditForm' => $companyForm,
+        ));
+
+        return $vm;
+    }
+    public function deletePacketAction()
+    {
+        $companyService = $this->getCompanyService();
+        $packetID = $this->params('packetID');
+        $companyName = $this->params('slugCompanyName');
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            $del = $request->getPost('del', 'No');
+            if ($del == 'Yes') {
```

Comment:
Please use === rather than ==, because otherwise it is a bit unpredictable when the company gets deleted